2-collectors/scala-stream-collector/core/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorRoute.scala

```diff
@@ -43,7 +42,7 @@ trait CollectorRoute {
           extractors { (host, ip, request) =>
             // get the adapter vendor and version from the path
             path(Segment / Segment) { (vendor, version) =>
-              val path = s"/$vendor/$version"
+              val path = collectorService.determinePath(vendor, version)
```

Comment:
So, does it mean we can rewrite only two-segmented URIs with two-segmented URIs? `/com.acme/foo/bar` won't be supported neither as a source nor destinatioin?

Reply:
Yes. This is the current behaviour as well.

Reply:
did you look into [Segments](https://doc.akka.io/docs/akka-http/current/routing-dsl/path-matchers.html) in the end to allow for arbitrary incoming paths?